### PR TITLE
Add boot sequence documentation

### PR DESCRIPTION
**Pull request by arnaldo2792**
**Issue number:**
Part of: https://github.com/bottlerocket-os/bottlerocket-project-website/issues/260

**Description of changes:**
This explains the systemd targets available in Bottlerocket, and a brief explanation of what happens in each of them. As part of this change, I updated the existing diagram and removed any mention of the boot sequence or systemd targets. I didn't close the existing issue because I want to evaluate if it makes sense to add a broader doc that explains in further detail each systemd service and how the system transitions from `preconfigured.target` -> `configured.target` -> `multi-user.target`.

**Testing done:**
You can look at the rendered diagram [here](https://github.com/arnaldo2792/bottlerocket/tree/systemd-docs/sources/api#api-system-components) and the new docs [here](https://github.com/arnaldo2792/bottlerocket/tree/systemd-docs/sources/api#boot-stages)

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by stmcginnis — APPROVED** [review]
Thanks for updating this!

**Review by zmrow — APPROVED** [review]
Nice work!
:robot:

File: sources/api/README.md [hunk 1]
```diff
@@ -143,12 +143,15 @@ Some day we may need to make an earlier service (say, apiserver) configurable th
...
+## Boot stages
 
-This is a systemd target that depends on [settings-applier](#settings-applier) and represents the point at which the system is fully configured.
+Bottlerocket's boot sequence includes three systemd targets:
 
-Applications can depend on this in their service definition.
-Services like Kubernetes and containerd depend on this.
+- `preconfigured.target`: this is the target where all the aforementioned services run.
+Once all the dynamically generated configurations are created, the system transitions to the next target
+- `configured.target`: this is the target that represents the point at which the system is fully configured, any configured [bootstrap container](../../bootstrap-containers-settings) will run at this stage to apply further configurations in the system.
```
Comment (webern) [inline sources/api/README.md]: Double use of the word "configured" seems confusing here. Unless there are two types of bootstrap containers (i.e. "configured bootstrap containers" and "some other type of bootstrap container") then I would recommend this:

```suggestion
- `configured.target`: this is the target that represents the point at which the system is fully configured.
   [Bootstrap containers](../../bootstrap-containers-settings) will run at this stage to apply further configurations in the system.
```

**Review by webern — APPROVED** [review]

**Comment by arnaldo2792** [comment]
(Forced push addresses comment by @webern)

**Comment by arnaldo2792** [comment]
(Another forced push to add a missing "s" in Bottstrap containers)

**Review by webern — APPROVED** [review]

File: sources/api/api-system.drawio [hunk 2]
```diff
@@ -1 +1 @@
...
...EIVPg4k4TiUtvgxHHjTZjNk0B/nu6FQDvNUhgf1wyrgV2B67jlMBVoXU10GZwdw6to0ALAcWb/pSHmD6PjzMUaJz70wzRmPF1Yhr/R0K8+SubN6W5l5+X78vRNvw2zVMS3FWsACIUQ8q/SUGLq+N5DDYjNSMhk7nRHHaDnH1ZRk6Dm3VK9ppqFFXxCv1rkdNFQMAgjpHHsYgZoEwVF9BLZtwh3Mi7mCypB5vTJdcXQNYc5aBfKi5UWxTQdjRgZzIKMWDovVyS6Cwgn/BEUMhyU5sVHzEq3E8/Wy4qVhAVPW6FMRU1KSqKmoQN228+gCBqfNuDICERCaCOHcZB7LB/S3b0t34u7TrYlx/9Skixj8eQh8GLNQE/v92NHvv0PwAerM9YW8y2IUiblNRofKel8d0zM77ZlfHdjzW+9ZHGd39Tz/9TjL9v7VBNAzXJo1A7wDViLzJTiOvvhevxunizOTifDP+MkGJ3xaojVhz6fGJ8ZEwx2xabZ27+QdVq+5pfiU7dmf9l4htfGb779v5jYt8Gq7GFnjXlxDIWmzgj7VjUdCq4qQHGEJOAggUnQAQp4m8iWhflsad84BQb5UptZhnqhu9Kwwqrg32yFt1T5etdpm10LeesXKsaVy8re/G2jmXb1bhaUdSdY83eVo9v1140cyer8ebx+set62kaVYrpRcMuak/7bf8fTDMNht5SctVgUNkTZ3ugpv6HedmBQ2hRaVHD1KCyE+T9oToJFNrEO+wWCvtgKLLDq5ZNsivjSMDoPEfp6MaQMRQGfF4AQ5ic3MR17VtlcfvO75ksvbi46KBbbTQnYR/E821nW9w8Acazd5hILGMg2rR8nzAimNBEvT1L/riccKWICRI5RrKnoOR1e5zI+XYjqIk8gK8xCoRCRiJtftvpLc21wCl4q82jVwpvI7xkRDFIVkTFDEaV2kmIfqFkmqE1zI6T9bXCgVjW5GRNGjE0SNvHihDq8YMC8p5ddA4c3SRb3wsnu/1eHMu3v8ld/f53V5Aulm67QmBj6SYteSalW3+r6OBNsWVUq7fT9uHV5BzzKEqCHdG1xVlgLEIkT1opZ4CMgx7nTOLtB2b62oze9tiri+NC7RvvX+p0Vgu7xp61sHusdGGqW29Z1MR9EEUYac5SD2BbS3K1zw3HJxe/zX85lHp2/vsr+/Z/</diagram></mxfile>
```
Comment (jpculp) [inline sources/api/api-system.drawio:1]: nit. missing line at end of file

Comment (arnaldo2792) [reply sources/api/api-system.drawio:1]: I didn't generate this file, it was generated by an IDE

**Pull request merged**